src/effects/VST/VSTEffect.h

```diff
@@ -282,6 +282,24 @@ struct VSTEffectWrapper : public VSTEffectLink, public XMLTagHandler
    // Some other methods called by the callback make sense for Instances:
    void         SetBufferDelay(int samples);
 
+
+   // The following is stuff that BuildFancy and BuildPlain use, but might
+   // be used by the Effect too - that's why they are here.
+   // TODO later: move as many as possible of these to the validator,
+   // possibly by creating copies of them in the Effect.
+   wxWindow* mParent;
+   wxWeakRef<wxDialog> mDialog;
+   ArrayOf<wxStaticText*> mNames;
+   ArrayOf<wxSlider*> mSliders;
+   ArrayOf<wxStaticText*> mDisplays;
+   ArrayOf<wxStaticText*> mLabels;
+   NumericTextCtrl* mDuration;
+   VSTControl* mControl;
+   void NeedEditIdle(bool state);
+      bool mWantsEditIdle{ false };
+      std::unique_ptr<VSTEffectTimer> mTimer;
+   void RefreshParameters(int skip = -1) const;
```

Comment:
These should not be in the Wrapper class.  Instead move all from the effect to the validator -- except that mDialog and mParent are duplicated in Effect and Validator.

Reply:
ok, done